solana-programs/reward-manager/program/tests/transfer.rs

```diff
@@ -264,8 +205,271 @@ async fn success_transfer() {
+        &claimable_tokens::id(),
+        &mint.pubkey(),
+        recipient_eth_key,
+    )
+    .unwrap();
+    create_recipient_with_claimable_program(&mut context, &mint.pubkey(), recipient_eth_key).await;
+
+    // attempt to evaluate the bad attestations
+    let tx = Transaction::new_signed_with_payer(
+        &[instruction::evaluate_attestations(
+            &audius_reward_manager::id(),
+            &verified_messages_account,
+            &reward_manager.pubkey(),
+            &token_account.pubkey(),
+            &recipient_sol_key.derive.address,
+            &oracle_derived_address,
+            &context.payer.pubkey(),
+            10_000u64,
+            transfer_id.to_string(),
+            recipient_eth_key,
+        )
+        .unwrap()],
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    let res = context.banks_client.process_transaction(tx).await;
+    // Ensure we got the expected NotEnoughSigners error
+    assert_custom_error(res, 0, AudiusProgramError::NotEnoughSigners);
+
+    // Ensure that the transfer account wasn't created
+    let transfer_account_data = get_account(&mut context, &transfer_account)
+        .await;
+    assert!(transfer_account_data.is_none());
+
+    // Try again to submit, this time with correct instructions
+    let mut instructions = Vec::<Instruction>::new();
+
+    // Only use operators 1-3
+    for item in keys[..3].iter().enumerate() {
+        let priv_key = SecretKey::parse(item.1).unwrap();
+        let inst = new_secp256k1_instruction_2_0(
+            &priv_key,
+            senders_message.as_ref(),
+            (2 * item.0) as u8,
+        );
+        instructions.push(inst);
+        instructions.push(
+            instruction::submit_attestations(
+                &audius_reward_manager::id(),
+                &reward_manager.pubkey(),
+                &signers[item.0],
+                &context.payer.pubkey(),
+                transfer_id.to_string()
+            )
+            .unwrap(),
+        );
+    }
+    // Now add the oracle
+    let oracle_sign = new_secp256k1_instruction_2_0(
+        &oracle_priv_key,
+        bot_oracle_message.as_ref(),
+        (instructions.len()) as u8,
+    );
+    instructions.push(oracle_sign);
+    instructions.push(
+        instruction::submit_attestations(
+            &audius_reward_manager::id(),
+            &reward_manager.pubkey(),
+            &oracle_derived_address,
+            &context.payer.pubkey(),
+            transfer_id.to_string()
+        )
+        .unwrap(),
+    );
+    println!("{:?}", instructions.len());
+
+    // Send the submit instruction
+    let tx = Transaction::new_signed_with_payer(
+        &instructions,
+        Some(&context.payer.pubkey()),
+        &[&context.payer],
+        context.last_blockhash,
+    );
+
+    context.banks_client.process_transaction(tx).await.unwrap();
```

Comment:
should we assert on the messages length here? before evaluate is submitted

Reply:
I think it's probably fine since the message account being in a good state is implicitly tested by evaluate working correclty